javascript/react/fundamentals/functional-components.md

```diff
@@ -35,27 +36,91 @@ aspects:
 
 As stated before, **React** components behave just like functions, taking `props` as input and returning **React elements**.
 
-Simple `component`s that don't have an internal `state` and don't make use of any **lifecycle** methods such as `constructor()` can be written as **functional components**.
+Up until recently, functional components couldn't make us of any **lifecycle** methods, nor could they access `this.state` or call `setState()`. With the introduction of hooks, there is no visible difference between functional and class components:
+
+```js
+function Enki() {
+  // defining the "name" state,
+  // a "setName()" function to change
+  // the state, and the initial
+  // state, 'Enki'.
+  const [name, setName] = useState("Enki");
+
+  useEffect(
+    // lifecycle methods
```

Comment:
This isn't totally correct. With `useEffect` we can achieve some lifecycle-like behavior but that's not the correct mental mapping, effects are a different mental model

Reply:
I've changed it to "perform side effects"